File: samples/typescript/src/tradeHistory.ts
import axios from 'axios';
import * as dotenv from 'dotenv';

dotenv.config();

const GEMINI_BASE_URL = process.env.GEMINI_BASE_URL || 'https://api.gemini.com/v1';

async function getTradeHistory(symbol: string): Promise<void> {
  try {
    const symbolLower = symbol.toLowerCase();
    const response = await axios.get(`${GEMINI_BASE_URL}/trades/${symbolLower}`);

    const trades = response.data;
    console.log(`Recent trades for ${symbol} (${trades.length} trades):\n`);

    for (const trade of trades) {
      const time = new Date(trade.timestampms).toISOString();
      console.log(`[${time}] ${trade.type.padEnd(4)} ${trade.amount} @ ${trade.price} (tid: ${trade.tid})`);
    }
  } catch (error) {
    console.error('Error fetching trade history:', error);
  }
}

// Get symbol from command line or use default
const symbol = process.argv[2] || 'btcusd';

if (process.argv.includes('--help') || process.argv.includes('-h')) {
  console.log('Usage: npx ts-node src/tradeHistory.ts [symbol]');
  console.log('Example: npx ts-node src/tradeHistory.ts ethusd');
  console.log('Default symbol: btcusd');
  process.exit(0);
}

getTradeHistory(symbol);

File: samples/python/trade_history.py
import os
import requests
from datetime import datetime, timezone
from dotenv import load_dotenv

load_dotenv()

GEMINI_BASE_URL = os.getenv('GEMINI_BASE_URL', 'https://api.gemini.com/v1')


def get_trade_history(symbol: str):
    """Fetch recent trade history for a given symbol."""
    symbol_lower = symbol.lower()
    url = f"{GEMINI_BASE_URL}/trades/{symbol_lower}"

    try:
        response = requests.get(url)
        response.raise_for_status()
        trades = response.json()

        print(f'Recent trades for {symbol} ({len(trades)} trades):\n')

        for trade in trades:
            time = datetime.fromtimestamp(trade['timestampms'] / 1000, tz=timezone.utc).isoformat()
            print(f"[{time}] {trade['type']:<4} {trade['amount']} @ {trade['price']} (tid: {trade['tid']})")
    except requests.exceptions.RequestException as e:
        print(f'Error fetching trade history: {e}')


if __name__ == '__main__':
    import sys

    # Show help message
    if '--help' in sys.argv or '-h' in sys.argv:
        print('Usage: python3 trade_history.py [symbol]')
        print('Example: python3 trade_history.py ethusd')
        print('Default symbol: btcusd')
        sys.exit(0)

    # Get symbol from command line or use default
    symbol = sys.argv[1] if len(sys.argv) > 1 else 'btcusd'
    get_trade_history(symbol)
